crates/shadowsocks/src/relay/udprelay/proxy_socket.rs

```diff
@@ -254,6 +251,76 @@ impl ProxySocket {
         Ok(send_len)
     }
 
+    pub fn poll_send(&self, addr: &Address, payload: &[u8], cx: &mut Context<'_>) -> Poll<ProxySocketResult<usize>> {
+        let mut send_buf = BytesMut::with_capacity(payload.len() + 256);
+
+        self.encrypt_send_buffer(
+            addr,
+            &DEFAULT_SOCKET_CONTROL,
+            &self.identity_keys,
+            payload,
+            &mut send_buf,
+        )?;
+
+        let n_send_buf = send_buf.len();
+
+        match self.socket.poll_send(cx, &mut send_buf.freeze()).map_err(|x| x.into()) {
+            Poll::Ready(Ok(l)) => {
+                if l == n_send_buf {
+                    Poll::Ready(Ok(payload.len()))
+                } else {
+                    Poll::Ready(Err(io::Error::from(ErrorKind::WriteZero).into()))
+                }
+            }
+            x => x,
+        }
+    }
+
+    pub fn poll_send_to(
```

Comment:
`poll_send_to_with_ctrl` and `poll_send_with_ctrl` are missing.

Reply:
updated